public/src/combinationalAnalysis.js

```diff
@@ -296,3 +320,146 @@ function drawCombinationalAnalysis(combinationalData, inputList, outputListNames
+                equation = equation.substring(0, i + 1) + '*' + equation.substring(i + 1, equation.length);
+        try {
+            let safeEval = eval;
+            let answer = safeEval(equation);
+            if (answer == 0)
+                return 0;
+            if (answer > 0)
+                return 1;
+            return '';
+        } catch (e) {
+            return '';
+        }
+    }
+ 
+    $('#combinationalAnalysis').empty();
+    $('#combinationalAnalysis').append(s);
+    $('#combinationalAnalysis').dialog({
+        resizable: false,
+        width: 'auto',
+        buttons: [
+            {
+                text: 'Generate Circuit',
+                click() {
+                    $(this).dialog('close');
+                    var data = generateBooleanTableData([13]);
+                    var minimizedCircuit = [];
+                    let inputCount = inputListNames.length;
+                    for (const output in data) {
+                        let oneCount = data[output][1].length;
+                        let zeroCount = data[output][0].length;
+                        if (oneCount == 0) {
+                            minimizedCircuit.push(['-'.repeat(inputCount) + '0']);
+                        }
+                        else if (zeroCount == 0) {
+                            minimizedCircuit.push(['-'.repeat(inputCount) + '1']);
+                        }
+                        else {
+                            const temp = new BooleanMinimize(
+                                inputListNames.length,
+                                data[output][1].map(Number),
+                                data[output].x.map(Number),
+                            );
+                            minimizedCircuit.push(temp.result);
+                        }
+                        
+                    }
+                    drawCombinationalAnalysis(minimizedCircuit, inputListNames, [`${booleanExpression}`], scope);
+                },
+            },
+            {
+                text: 'Print Truth Table',
+                click() {
+                    var sTable = document.getElementById('combinationalAnalysis').innerHTML;
+                    var style = '<style> table {font: 20px Calibri;} table, th, td {border: solid 1px #DDD;border-collapse: collapse;} padding: 2px 3px;text-align: center;} </style>';
+                    var win = window.open('', '', 'height=700,width=700');
+                    win.document.write('<html><head>');
+                    win.document.write('<title>Boolean Logic Table</title>');
+                    win.document.write(style);
+                    win.document.write('</head>');
+                    win.document.write('<body>');
+                    win.document.write(`<center>${sTable}</center>`);
+                    win.document.write('</body></html>');
+                    win.document.close();
```

Comment:
Since document write can be expensive, can you build the whole string and write to dom ? 

htmlBody += <html><head>
htmlBody += 
htmlBody += 
htmlBody += 
htmlBody += 
document.write(htmlBody)

Reply:
made changes accordingly pls have a look on it !!

Reply:
Alternatively use multiline string @pavanjoshi914 , it will look better

Reply:
okay making changes in latest commits !!